packages/supersearch/src/lib/components/SuperSearch.svelte

```diff
@@ -1,12 +1,19 @@
 <script lang="ts">
 	import CodeMirror, { type ChangeCodeMirrorEvent } from '$lib/components/CodeMirror.svelte';
 	import { placeholder } from '@codemirror/view';
+	import { type LRLanguage } from '@codemirror/language';
 
 	let value = $state('');
+	let { language }: { language?: LRLanguage } = $props();
+
+	const extensions = [placeholder('Search')];
+	if (language) {
+		extensions.push(language);
+	}
```

Comment:
It's a matter of taste but I prefer conditionally adding the extensions inline [using the spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#conditionally_adding_properties_to_an_object), so you don’t have to be as aware of the code around in the same extent: 

```js
	const extensions = [
                placeholder('Search'),
		...(language ? [language] : [])
	]
```

For me it becomes clearer where all the initially assigned properties come from without having to look through the whole file (vs a quick glance at the assignment) – it isn't a problem right now but can be when the list of extensions grow.

The conditional spread will also play along better with the usage of [compartments](https://codemirror.net/examples/config/#compartments) for dynamic partial reconfiguration, as the original list of extension should be treated more or less as an constant (_it’s not the extensions that changes, it’s the compartments_) to prevent the costly reconfiguration of all extensions.

Reply:
Good point, fixed now.